cypress/tests/ui/flowchart/shareable-urls.cy.js

```diff
@@ -34,4 +34,182 @@ describe('Shareable URLs', () => {
+
+    // Assert after action
+    cy.get('.shareable-url-modal .menu-option').should(
+      'have.length',
+      regionCount
+    );
+  });
+
+  it('verifies that publish button should be disabled when region is not selected and bucket name is empty #TC-56', () => {
+    const selectedRegion = 'Choose a region...';
+    const primaryButtonNodeText = 'Publish';
+
+    // Action
+    cy.get('.pipeline-menu-button--deploy').click();
+
+    // Assert after action
+    cy.get(
+      '.shareable-url-modal [data-test=kedro-pipeline-selector] .dropdown__label span'
+    ).contains(selectedRegion);
+    cy.get('.shareable-url-modal textarea').should('have.value', '');
+    cy.get('.shareable-url-modal__button-wrapper button')
+      .contains(primaryButtonNodeText)
+      .should('be.disabled');
+  });
+
+  it('verifies that publish button should be disabled when region is not selected or bucket name is empty #TC-57', () => {
+    const primaryButtonNodeText = 'Publish';
+
+    // Action
+    cy.get('.pipeline-menu-button--deploy').click();
+    cy.get('.shareable-url-modal [data-test=kedro-pipeline-selector]').click();
+    cy.get('.shareable-url-modal .dropdown__options section div')
+      .first()
+      .click();
+
+    // Assert after action
+    cy.get('.shareable-url-modal textarea').should('have.value', '');
+    cy.get('.shareable-url-modal__button-wrapper button')
+      .contains(primaryButtonNodeText)
+      .should('be.disabled');
+  });
+
+  it('verifies that publish button should be enabled when region is selected and bucket name is not empty #TC-58', () => {
+    const bucketName = 'myBucketName';
+    const primaryButtonNodeText = 'Publish';
+
+    // Action
+    cy.get('.pipeline-menu-button--deploy').click();
+    cy.get('.shareable-url-modal [data-test=kedro-pipeline-selector]').click();
+    cy.get('.shareable-url-modal .dropdown__options section div')
+      .first()
+      .click();
+    cy.get('.shareable-url-modal textarea').type(bucketName);
+
+    // Assert after action
+    cy.get('.shareable-url-modal__button-wrapper button')
+      .contains(primaryButtonNodeText)
+      .should('be.enabled');
+  });
+
+  it('verifies that error message appears with wrong inputs on publish button click #TC-59', () => {
+    const bucketName = 'myBucketName';
+    const primaryButtonNodeText = 'Publish';
+
+    // Action
+    cy.get('.pipeline-menu-button--deploy').click();
+    cy.get('.shareable-url-modal [data-test=kedro-pipeline-selector]').click();
+    cy.get('.shareable-url-modal .dropdown__options section div')
+      .first()
+      .click();
+    cy.get('.shareable-url-modal textarea').type(bucketName);
+    cy.get('.shareable-url-modal__button-wrapper button')
+      .contains(primaryButtonNodeText)
+      .click();
+
+    // Assert after action
+    cy.get('.shareable-url-modal .modal__wrapper').contains(
+      'Something went wrong. Please try again later.'
+    );
```

Comment:
I've added a `Go back` button to this state. Could you please check that it also exists here?